lib/utils/azure_queues.ts

```diff
@@ -127,3 +135,68 @@ export function updateMessageVisibilityTimeout<T extends IQueueMessage>(
       });
   });
 }
+
+/**
+ * Returns false when we need to trigger a retry.
+ */
+export async function handleQueueProcessingFailure(
+  queueService: QueueService,
+  queueMessage: IQueueMessage,
+  queueName: string,
+  onTransientError: () => Promise<Either<RuntimeError, {}>>,
+  onPermanentError: () => Promise<Either<RuntimeError, {}>>,
+  error: Error | RuntimeError
```

Comment:
this is quite confusing, this function accepts an error of type `Error` but it handles it as a `PermanentError`, that is a subtype of  `RuntimeError` - I suggest to just accept a `RuntimeError` here

Reply:
we may do that, but the the caller is responsible to translate an Error in a RunTimeError (calling RuntimeErrorOf)